drivers/serial/uart_npcx.c

```diff
@@ -57,20 +69,22 @@ struct uart_npcx_data {
 	(struct uart_reg *)(DRV_CONFIG(dev)->uconf.base)
 
 #if defined(CONFIG_PM) && defined(CONFIG_UART_INTERRUPT_DRIVEN)
-static void uart_npcx_pm_constraint_set(const struct device *dev)
+static void uart_npcx_pm_constraint_set(const struct device *dev,
```

Comment:
If you change the parameter of the constraint helper routines to `struct uart_npcx_data *data`, then you can eliminate the `data->uart_dev` field.

Reply:
Done.